modules/ding_provider/connie_search/src/ConnieTingObjectCollection.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @file
+ * The ConnieTingObjectCollection class.
+ */
+
+namespace Connie\Search;
+
+use Ting\TingObjectCollectionInterface;
+
+class ConnieTingObjectCollection implements TingObjectCollectionInterface {
+
+  /**
+   * @var \Ting\TingObjectInterface[]
+   */
+  protected $objects;
+
+  /**
+   * ConnieObjectCollection constructor.
+   *
+   * @param \Ting\TingObjectInterface[] $objects
+   *   The objects in the collection.
+   */
+  public function __construct($objects) {
+    $this->objects = $objects;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getObjects() {
+    return $this->objects;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPrimaryObject() {
+    // Get the first array-entry (without modifying the array).
+    $objects = array_values($this->objects);
```

Comment:
Two things
1: It resets the internal array-pointe
2: You have to know that reset() has the wird side-effect of returning the first element after the reset

Not anything huge, but I think this code reads better.